packages/web-components/src/avatar/avatar.ts

```diff
@@ -0,0 +1,175 @@
+import { attr, FASTElement, nullableNumberConverter } from '@microsoft/fast-element';
+import { getInitials } from '../utils/get-initials.js';
+import {
+  AvatarActive,
+  AvatarAppearance,
+  AvatarColor,
+  AvatarNamedColor,
+  AvatarShape,
+  AvatarSize,
+} from './avatar.options.js';
+
+/**
+ * The base class used for constructing a fluent-avatar custom element
+ * @public
+ */
+export class Avatar extends FASTElement {
+  /**
+   * The name of the person or entity represented by this Avatar. This should always be provided if it is available.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: name
+   */
+  @attr
+  public name: string | undefined;
+  protected nameChanged(previous: string, next: string): void {
+    if (next !== undefined && !this.initials) {
+      this.setInitials();
+    }
+  }
+
+  /**
+   * Provide custom initials rather than one generated via the name
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: name
+   */
+  @attr
+  public initials: string | undefined;
+  protected initialsChanged(previous: string | undefined, next: string | undefined): void {
+    if (next !== undefined) {
+      this.setInitials();
+    }
+  }
+
+  /**
+   * Size of the avatar in pixels.
+   *
+   * Size is restricted to a limited set of supported values recommended for most uses (see `AvatarSizeValue`) and
+   * based on design guidelines for the Avatar control.
+   *
+   * If a non-supported size is neeeded, set `size` to the next-smaller supported size, and set `width` and `height`
+   * to override the rendered size.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: size
+   *
+   */
+  @attr({ converter: nullableNumberConverter })
+  public size: AvatarSize | undefined;
+
+  /**
+   * The avatar can have a circular or square shape.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: shape
+   */
+  @attr
+  public shape: AvatarShape | undefined;
+
+  /**
+   * Optional activity indicator
+   * * active: the avatar will be decorated according to activeAppearance
+   * * inactive: the avatar will be reduced in size and partially transparent
+   * * undefined: normal display
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: active
+   */
+  @attr
+  public active?: AvatarActive | undefined;
```

Comment:
I think this is the only outstanding instance, I'm going to look for detail on this in past issues.

Reply:
OK, I've found what I think is precedent from FUIR9 for adopting `undefined` here instead of "unset". This specific instance seems to be a large part of prompting this RFC: https://github.com/microsoft/fluentui/pull/23304. Given we're doing new work here, I'd rather align with this forward-thinking direction which I assume will be adopted into the React Avatar implementation when we can next make a breaking change.

Reply:
Makes sense, let's go with `undefined`. Do you plan to add a spec documenting the differences between FUIR9 and WC3?